util/export_mtz.py

```diff
@@ -125,135 +125,107 @@ def add_dataset(
         )
 
 
-def _add_batch(
+def _add_batch_list(
+    image_range,
     mtz,
     experiment,
     wavelength,
     dataset_id,
-    batch_number,
-    image_number,
+    batch_offset,
     force_static_model,
 ):
-    """Add a single image's metadata to an mtz file.
-
-    Returns the batch object.
-    """
-    assert batch_number > 0
+    """Add batch metadata to an mtz file."""
 
     # Recalculate useful numbers and references here
-    # We ignore panels beyond the first one, at the moment
-    panel = experiment.detector[0]
-
-    if experiment.goniometer:
-        axis = matrix.col(experiment.goniometer.get_rotation_axis())
-    else:
-        axis = 0.0, 0.0, 0.0
+    n_batches = image_range[1] - image_range[0] + 1
+    phi_start = flex.float(n_batches)
+    phi_range = flex.float(n_batches)
+    umat_array = flex.float(flex.grid(n_batches, 9))
+    cell_array = flex.float(flex.grid(n_batches, 6))
 
     U = matrix.sqr(experiment.crystal.get_U())
     if experiment.goniometer is not None:
         F = matrix.sqr(experiment.goniometer.get_fixed_rotation())
     else:
         F = matrix.sqr((1, 0, 0, 0, 1, 0, 0, 0, 1))
 
-    # Create the batch object and start configuring it
-    o = mtz.add_batch().set_num(batch_number).set_nbsetid(dataset_id).set_ncryst(1)
-    o.set_time1(0.0).set_time2(0.0).set_title("Batch {}".format(batch_number))
-    o.set_ndet(1).set_theta(flex.float((0.0, 0.0))).set_lbmflg(0)
-    o.set_alambd(wavelength).set_delamb(0.0).set_delcor(0.0)
-    o.set_divhd(0.0).set_divvd(0.0)
+    i0 = image_range[0]
+    for i in range(n_batches):
+        if experiment.scan:
+            phi_start[i], phi_range[i] = experiment.scan.get_image_oscillation(i + i0)
+        else:
+            phi_start[i], phi_range[i] = 0.0, 0.0
```

Comment:
is `flex.float()` guaranteed to be 0-initialised? If so could drop the `else:` clause

Reply:
@Anthchirp by default no but we can force it to be... good suggestion. `flex.float(size, init_value)` will do it